### ISSUE TITLE
Image resize error

### ISSUE THREAD

**Issue by tphlrussia** [Deci-AI/super-gradients#1334]
### 🐛 Describe the bug

When using yolo_nas_m with --half argument via track.py

File "/home/timur/miniconda3/envs/bo233/lib/python3.10/site-packages/super_gradients/training/pipelines/pipelines.py", line 133, in _generate_prediction_result
    return [result for batch_images in generate_batch(images, batch_size) for result in self._generate_prediction_result_single_batch(batch_images)]
  File "/home/timur/miniconda3/envs/bo233/lib/python3.10/site-packages/super_gradients/training/pipelines/pipelines.py", line 133, in <listcomp>
    return [result for batch_images in generate_batch(images, batch_size) for result in self._generate_prediction_result_single_batch(batch_images)]
  File "/home/timur/miniconda3/envs/bo233/lib/python3.10/site-packages/super_gradients/training/pipelines/pipelines.py", line 152, in _generate_prediction_result_single_batch
    preprocessed_image, processing_metadata = self.image_processor.preprocess_image(image=image.copy())
  File "/home/timur/miniconda3/envs/bo233/lib/python3.10/site-packages/super_gradients/training/processing/processing.py", line 74, in preprocess_image
    processed_image, metadata = processing.preprocess_image(image=processed_image)
  File "/home/timur/miniconda3/envs/bo233/lib/python3.10/site-packages/super_gradients/training/processing/processing.py", line 243, in preprocess_image
    image = _rescale_image(image, target_shape=(new_height, new_width))
  File "/home/timur/miniconda3/envs/bo233/lib/python3.10/site-packages/super_gradients/training/transforms/utils.py", line 26, in _rescale_image
    return cv2.resize(image, dsize=(width, height), interpolation=cv2.INTER_LINEAR).astype(np.uint8)
cv2.error: OpenCV(4.7.0) :-1: error: (-5:Bad argument) in function 'resize'
> Overload resolution failed:
>  - src data type = 23 is not supported
>  - Expected Ptr<cv::UMat> for argument 'src'

As I think   the data type of the image argument passed to the cv2.resize function is not supported. 

Versions:
python=3.10.8
super-gradients==3.1.0
torch==1.13
torchvision==0.14.0
ultralytics==8.0.124
opencv-python==4.7.0.72
numpy==1.23

In fact, I'm not sure that this error  is on your side, your package is used via boxmot

### Versions

Collecting environment information...
PyTorch version: 1.13.0+cu117
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Fedora Linux 38 (Workstation Edition) (x86_64)
GCC version: (GCC) 13.1.1 20230614 (Red Hat 13.1.1-4)
Clang version: Could not collect
CMake version: version 3.27.0
Libc version: glibc-2.37

Python version: 3.10.8 | packaged by conda-forge | (main, Nov 22 2022, 08:26:04) [GCC 10.4.0] (64-bit runtime)
Python platform: Linux-6.3.12-200.fc38.x86_64-x86_64-with-glibc2.37
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce GTX 1060 6GB
Nvidia driver version: 535.54.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Архитектура:                     x86_64
CPU op-mode(s):                  32-bit, 64-bit
Address sizes:                   39 bits physical, 48 bits virtual
Порядок байт:                    Little Endian
CPU(s):                          6
On-line CPU(s) list:             0-5
ID прроизводителя:               GenuineIntel
Имя модели:                      Intel(R) Core(TM) i5-9400F CPU @ 2.90GHz
Семейство ЦПУ:                   6
Модель:                          158
Thread(s) per core:              1
Ядер на сокет:                   6
Сокетов:                         1
Степпинг:                        10
CPU(s) scaling MHz:              28%
CPU max MHz:                     4100,0000
CPU min MHz:                     800,0000
BogoMIPS:                        5799,77
Флаги:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb invpcid_single pti ssbd ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid mpx rdseed adx smap clflushopt intel_pt xsaveopt xsavec xgetbv1 xsaves dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp md_clear flush_l1d arch_capabilities
Виртуализация:                   VT-x
L1d cache:                       192 KiB (6 instances)
L1i cache:                       192 KiB (6 instances)
L2 cache:                        1,5 MiB (6 instances)
L3 cache:                        9 MiB (1 instance)
NUMA node(s):                    1
NUMA node0 CPU(s):               0-5
Vulnerability Itlb multihit:     KVM: Mitigation: VMX disabled
Vulnerability L1tf:              Mitigation; PTE Inversion; VMX conditional cache flushes, SMT disabled
Vulnerability Mds:               Mitigation; Clear CPU buffers; SMT disabled
Vulnerability Meltdown:          Mitigation; PTI
Vulnerability Mmio stale data:   Mitigation; Clear CPU buffers; SMT disabled
Vulnerability Retbleed:          Mitigation; IBRS
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; IBRS, IBPB conditional, STIBP disabled, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:             Mitigation; Microcode
Vulnerability Tsx async abort:   Not affected

Versions of relevant libraries:
[pip3] numpy==1.23.0
[pip3] torch==1.13.0
[pip3] torchmetrics==0.8.0
[pip3] torchvision==0.14.0
[pip3] triton==2.0.0
[conda] libopenvino-pytorch-frontend 2023.0.1             h59595ed_2    conda-forge
[conda] numpy                     1.23.0                   pypi_0    pypi
[conda] torch                     1.13.0                   pypi_0    pypi
[conda] torchmetrics              0.8.0                    pypi_0    pypi
[conda] torchvision               0.14.0                   pypi_0    pypi
[conda] triton                    2.0.0                    pypi_0    pypi
